### PR TITLE
TextLintBearTest: Remove unnecessary result

### PR DESCRIPTION
**Pull request by jayvdb**
The test data for test_bad_alex_no_dead_link was causing
a error about gender, which was unnecessary in this test.
The underlying error has changed slightly in newer versions
of the linter, so removing the test case allows the bear
to pass tests on a wider array of linter versions.

Closes https://github.com/coala/coala-bears/issues/2777

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: tests/general/textlint_test_files/bad_alex_no_dead_link.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
-We've confirmed his identity.
+We've confirmed their identity.
```
Comment (li-boxuan) [inline tests/general/textlint_test_files/bad_alex_no_dead_link.md]: Since this is unrelated to the `no_dead_link` test, why not remove it?

**Comment by li-boxuan** [comment]
A typo in the commit message:
`was causing a error about gender` -> `was causing an error about gender`

File: tests/general/textlint_test_files/bad_alex_no_dead_link.md (hunk 1 above)
Comment (jayvdb) [reply tests/general/textlint_test_files/bad_alex_no_dead_link.md]: filling the line allows the line numbers of the other result to stay the same.

It is possible for any content on this line to become an error in the future, as textlint adds more rules, so the text on this line is as good as any other.

**Comment by jayvdb** [comment]
@gitmate-bot ff
